File: web/src/api/wallet.ts
// web/src/api/wallet.ts
import request from './request'

// ========== 类型定义 ==========

export interface UserWallet {
  id: number
  user_id: number
  balance: number
  frozen_balance: number
  total_income: number
  total_spent: number
  created_at: string
  updated_at: string
}

export interface WalletTransaction {
  id: number
  user_id: number
  type: string
  amount: number
  balance: number
  ref_type: string
  ref_id: number
  description: string
  created_at: string
}

// 流水类型标签
export const txTypeLabels: Record<string, string> = {
  recharge: '充值',
  purchase: '购买',
  income: '收入',
  withdraw: '提现',
  refund: '退款',
}

// ========== API ==========

export const walletApi = {
  // 钱包信息
  getWallet: () =>
    request.get('/wallet'),

  // 流水列表
  listTransactions: (params?: { page?: number; page_size?: number }) =>
    request.get('/wallet/transactions', { params }),

  // 充值
  recharge: (data: { user_id: number; amount: number }) =>
    request.post('/wallet/recharge', data),
}

File: server/internal/model/wallet.go
// server/internal/model/wallet.go
package model

import "time"

// UserWallet 用户钱包
type UserWallet struct {
	ID            uint      `gorm:"primaryKey" json:"id"`
	UserID        uint      `gorm:"uniqueIndex;not null" json:"user_id"`
	Balance       int64     `gorm:"default:0;not null" json:"balance"`
	FrozenBalance int64     `gorm:"default:0;not null" json:"frozen_balance"`
	TotalIncome   int64     `gorm:"default:0;not null" json:"total_income"`
	TotalSpent    int64     `gorm:"default:0;not null" json:"total_spent"`
	Version       int64     `gorm:"default:0;not null" json:"-"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

// WalletTransaction 钱包流水
type WalletTransaction struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	UserID      uint      `gorm:"index;not null" json:"user_id"`
	Type        string    `gorm:"size:20;not null" json:"type"`
	Amount      int64     `gorm:"not null" json:"amount"`
	Balance     int64     `gorm:"not null" json:"balance"`
	RefType     string    `gorm:"size:30" json:"ref_type"`
	RefID       uint      `gorm:"default:0" json:"ref_id"`
	Description string    `gorm:"size:200" json:"description"`
	CreatedAt   time.Time `json:"created_at"`
}

// 流水类型常量
const (
	TxTypeRecharge      = "recharge"
	TxTypePurchase      = "purchase"
	TxTypeIncome        = "income"
	TxTypeWithdraw      = "withdraw"
	TxTypeRefund        = "refund"
	TxTypeLevelPurchase = "level_purchase" // 大神写手解锁购买
	TxTypeTokenConsume  = "token_consume"  // Token 消耗扣费
)
